dht_test.go

```diff
@@ -126,17 +127,19 @@ func connectNoSync(t *testing.T, ctx context.Context, a, b *IpfsDHT) {
 	}
 }
 
-func connect(t *testing.T, ctx context.Context, a, b *IpfsDHT) {
+func connect(t *testing.T, ctx context.Context, maxWaitMs uint64, a, b *IpfsDHT) {
 	connectNoSync(t, ctx, a, b)
 
 	// loop until connection notification has been received.
 	// under high load, this may not happen as immediately as we would like.
-	for a.routingTable.Find(b.self) == "" {
+	for maxWaitMs > 0 && a.routingTable.Find(b.self) == "" {
 		time.Sleep(time.Millisecond * 5)
+		maxWaitMs -= 5
 	}
 
-	for b.routingTable.Find(a.self) == "" {
```

Comment:
We should probably fail the test if this connection fails.

Also, were you noticing a test failure here?

Reply:
> We should probably fail the test if this connection fails.

Interesting. The new test that I wrote will not be able to use this function, then, because A's routing table will never include B if A and B are configured to use different protocols. That should be fine. I'll just connect A and B's hosts, sleep for 2 seconds, and then run my Get/Put test.

How's that sound?

Reply:
> Also, were you noticing a test failure here?

In the new test that I wrote (which uses different protocols for A and B), I did not see a failure. Rather, the timeout expired and the polling simply stops and the function returns.